oracle/metrics/metrics.go

```diff
@@ -71,113 +85,100 @@ type OracleMetricsImpl struct {
 // config.
 func NewMetricsFromConfig(config config.MetricsConfig) Metrics {
 	if config.Enabled {
-		return NewMetrics()
+		var telemetryPushAddress string
+		if !config.Telemetry.Disabled {
+			telemetryPushAddress = config.Telemetry.PushAddress
+		}
+		return NewMetrics(telemetryPushAddress)
 	}
 	return NewNopMetrics()
 }
 
+func NewNopMetrics() Metrics {
+	return &OracleMetricsImpl{}
+}
+
 // NewMetrics returns a Metrics implementation that exposes metrics to Prometheus.
-func NewMetrics() Metrics {
-	m := &OracleMetricsImpl{
+func NewMetrics(telemetryPushAddress string) Metrics {
+	var statsdClient *statsd.Client
+	if telemetryPushAddress != "" {
+		statsdClient, _ = statsd.New(telemetryPushAddress)
+	}
+	promMetrics := &PromMetrics{
 		ticks: prometheus.NewCounter(prometheus.CounterOpts{
 			Namespace: OracleSubsystem,
-			Name:      "health_check_system_updates_total",
+			Name:      TicksMetricName,
 			Help:      "Number of ticks with a successful oracle update.",
 		}),
 		tickerTicks: prometheus.NewCounterVec(prometheus.CounterOpts{
 			Namespace: OracleSubsystem,
-			Name:      "health_check_ticker_updates_total",
+			Name:      TickerTicksMetricName,
 			Help:      "Number of ticks with a successful ticker update.",
 		}, []string{PairIDLabel}),
 		prices: prometheus.NewGaugeVec(prometheus.GaugeOpts{
 			Namespace: OracleSubsystem,
-			Name:      "provider_price",
+			Name:      PricesMetricName,
 			Help:      "Price gauge for a given currency pair on a provider",
 		}, []string{ProviderLabel, PairIDLabel, DecimalsLabel}),
 		aggregatePrices: prometheus.NewGaugeVec(prometheus.GaugeOpts{
 			Namespace: OracleSubsystem,
-			Name:      "aggregated_price",
+			Name:      AggregatePricesMetricName,
 			Help:      "Aggregate price for a given currency pair",
 		}, []string{PairIDLabel, DecimalsLabel}),
 		providerTick: prometheus.NewCounterVec(prometheus.CounterOpts{
 			Namespace: OracleSubsystem,
-			Name:      "health_check_provider_updates_total",
+			Name:      ProviderTickMetricName,
 			Help:      "Number of ticks with a successful provider update.",
 		}, []string{ProviderLabel, PairIDLabel, SuccessLabel}),
 		providerCount: prometheus.NewGaugeVec(prometheus.GaugeOpts{
-			Namespace: OracleSubsystem,
-			Name:      "health_check_market_providers",
+			Namespace: ProviderCountMetricName,
+			Name:      ProviderTickMetricName,
 			Help:      "Number of providers that were utilized to calculate the final price for a given market.",
 		}, []string{PairIDLabel}),
 		slinkyBuildInfo: prometheus.NewGaugeVec(prometheus.GaugeOpts{
 			Namespace: OracleSubsystem,
-			Name:      "slinky_build_info",
+			Name:      SlinkyBuildInfoMetricName,
 			Help:      "Information about the slinky build",
```

Comment:
Can we make an issue to deprecate this metric and move to a `connect_build_info`

Reply:
done